content/getting-started/solving-kata.md

```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
 
 # Solving Kata
 
-<!--
-TODO Trainer, submitting a solution
-TODO Solution troubleshooting
-TODO forfeiting a kata
-TODO reward: honor, progress, privileges
-TODO satisfaction voting
--->
+After clicking on a link to a kata, you are presented with the [kata details (TODO: link to kata details panel doc)]() view with general information about it. Read carefully through the description, and if you are ready to face the challenge, click `TRAIN`.
+
+## Kata Trainer
+
+Welcome to the kata trainer! You use this view to take your attempts on solution, write the code and run tests. A few tips to get you started:
+
+### Solution editor
+
+You use solution editor to write code which solves the task. You can type the code straight into the editor, but if you want, you can use your favorite IDE and just copy your solution from it and paste it here.
```

Comment:
> About "when the tests fail over again" and "unlocking solutions":
> 
> * unlocking might lay into the current page because of the related button. But then I'd put it after the "submit final" section. Note: `## Unlocking solutions`, in this case, not `# Un...`
> * the trouble shooting part doesn't really feel right in this page. It's related to solving, yeah, for sure. I feel like if it's the last section of the page it's ok (then: submit final -> unlock -> trouble shooting -> end)

I was trying to follow the "natural" flow of the way how newbie would go through a kata:
1.  write solution,
2. Run sample tests
3. Run full tests, which potentially fail many times,
4. Go search for help,
5. Unlock solutions when nothing helps 

Order of headers follows this process, which I think is natural and different order seems a bit off to me.